icechunk-python/python/icechunk/__init__.py

```diff
@@ -350,6 +350,30 @@ async def async_merge(self, changes: bytes) -> None:
         """
         return await self._store.async_merge(changes)
 
+    def rebase(self, solver: ConflictDetector | BasicConflictSolver) -> None:
```

Comment:
I wonder if we should write a few paragraphs of design document on this. Two topics would be:
* how much do we want to expose of the solvers? This methods for example requires adding here when different strategies are created, and doesn't allow for users to extend the mechanisms  (which I'd love some day)
* how do we expose the results of the conflict detector. This is going to be important, I expect most production jobs to at least report to logs the conflicts when they find them.